### PR TITLE
doc: Updated release notes for Matter

### PR DESCRIPTION
**Pull request by kkasperczyk-no**
Updated release notes related to Matter for nRF Connect SDK 2.3.0.

### CODE REVIEW

**Review by LuDuda — APPROVED** [review]

**Review by greg-fer — APPROVED** [review]

File: doc/nrf/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -586,10 +584,21 @@ Matter samples
...
 * :ref:`matter_light_switch_sample`:
 
   * Added Wi-Fi low power configuration using Wi-Fi's :ref:`Legacy Power Save mode <ug_nrf70_developing_powersave_dtim_unicast>`.
 
+* :ref:`matter_light_bulb_sample`:
+
+  * Positively verified against ``Works with Google`` certification tests.
+  * Tested compatibility with:
+    * Google Home ecosystem for both Matter over Thread and Matter over Wi-Fi solutions (tested with Google Nest Hub 2nd generation for software version: 47.9.4.447810048 and Chromecast firmware version: 1.56.324896, and Google Home mobile application 2.63.1.12 version).
+    * Apple Home ecosystem for both Matter over Thread and Matter over Wi-Fi solutions (tested with Apple HomePod mini and Apple iPhone for iOS 16.3 version).
```
Comment (LuDuda) [inline doc/nrf/releases/release-notes-changelog.rst:598]:
```suggestion
    * Apple Home ecosystem for both Matter over Thread and Matter over Wi-Fi solutions (tested with Apple HomePod mini and Apple iPhone with iOS 16.3 version).
```

**Review by LuDuda — APPROVED** [review]

File: doc/nrf/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -586,10 +584,21 @@ Matter samples
       See :ref:`matter_lock_sample_wifi_thread_switching` in the sample documentation for more information.
     * Wi-Fi low power configuration using Wi-Fi's :ref:`Legacy Power Save mode <ug_nrf70_developing_powersave_dtim_unicast>`.
 
+  * Positively verified against ``Works with Google`` certification tests.
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-changelog.rst:587]:
```suggestion
  * Positively verified against "Works with Google" certification tests. 
```
It's not a code name or parameter, so no point in using ``.

File: doc/nrf/releases/release-notes-changelog.rst [hunk 3]
```diff
@@ -586,10 +584,21 @@ Matter samples
...
 
+  * Positively verified against ``Works with Google`` certification tests. 
+
 * :ref:`matter_light_switch_sample`:
 
   * Added Wi-Fi low power configuration using Wi-Fi's :ref:`Legacy Power Save mode <ug_nrf70_developing_powersave_dtim_unicast>`.
 
+* :ref:`matter_light_bulb_sample`:
+
+  * Positively verified against ``Works with Google`` certification tests.
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-changelog.rst:595]:
```suggestion
  * Positively verified against "Works with Google" certification tests.
```
Idem.

File: doc/nrf/releases/release-notes-changelog.rst [hunk 4]
```diff
@@ -586,10 +584,21 @@ Matter samples
...
+  * Tested compatibility with:
+    * Google Home ecosystem for both Matter over Thread and Matter over Wi-Fi solutions (tested with Google Nest Hub 2nd generation for software version: 47.9.4.447810048 and Chromecast firmware version: 1.56.324896, and Google Home mobile application 2.63.1.12 version).
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-changelog.rst:597]:
```suggestion
  * Tested compatibility with the following ecosystems:
  
    * Google Home ecosystem for both Matter over Thread and Matter over Wi-Fi solutions.
       Tested with Google Nest Hub 2nd generation (software version: 47.9.4.447810048; Chromecast firmware version: 1.56.324896, and Google Home mobile application v2.63.1.12).
```

File: doc/nrf/releases/release-notes-changelog.rst (hunk 1 above)
Comment (greg-fer) [inline doc/nrf/releases/release-notes-changelog.rst:598]:
```suggestion
    * Apple Home ecosystem for both Matter over Thread and Matter over Wi-Fi solutions.
      Tested with Apple HomePod mini and Apple iPhone (iOS v16.3).
```

File: doc/nrf/releases/release-notes-changelog.rst [hunk 5]
```diff
@@ -586,10 +584,21 @@ Matter samples
...
+    * Apple Home ecosystem for both Matter over Thread and Matter over Wi-Fi solutions (tested with Apple HomePod mini and Apple iPhone for iOS 16.3 version).
+    * Samsung SmartThings ecosystem for Matter over Thread solution (tested with Aeotec Smart Home Hub and SmartThings mobile application 1.7.97.22 version).
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-changelog.rst:599]:
```suggestion
    * Samsung SmartThings ecosystem for Matter over Thread solution.
      Tested with Aeotec Smart Home Hub and SmartThings mobile application (v1.7.97.22).
```

File: doc/nrf/releases/release-notes-changelog.rst [hunk 6]
```diff
@@ -586,10 +584,21 @@ Matter samples
...
+    * Amazon Alexa ecosystem for both Matter over Thread and Matter over Wi-Fi solutions (tested with Amazon Echo Dot and Amazon Alexa mobile application 2.2.495949.0 version).
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-changelog.rst:600]:
```suggestion
    * Amazon Alexa ecosystem for both Matter over Thread and Matter over Wi-Fi solutions.
      Tested with Amazon Echo Dot and Amazon Alexa mobile application (v2.2.495949.0).
```

**Review by greg-fer — CHANGES_REQUESTED** [review]

**Pull request closed without merging**